gremlin-go/driver/gorillaTransporter.go

```diff
@@ -0,0 +1,91 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package gremlingo
+
+import (
+	"net/url"
+	"strconv"
+
+	"github.com/gorilla/websocket"
+)
+
+type gorillaTransporter struct {
+	host       string
+	port       int
+	connection websocketConn
+	isClosed   bool
+}
+
+func (transporter *gorillaTransporter) Connect() (err error) {
+	if transporter.connection != nil {
+		return
```

Comment:
please return an error so that caller can know it is a non-retriable error and also that it is a configuration error.

Reply:
This part of the logic has been refactored as of MS2 to have a more intuitive workflow.